### PR TITLE
Convert breaking change template to a form

### PR DESCRIPTION
**Pull request by gewarren**
See https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/breaking-change.md [hunk 1]
```diff
@@ -1,65 +1,80 @@
...
-]
-
-<!-- Do not modify anything below this line -->
-
----
-#### Issue metadata
-
-* Issue type: breaking-change
+name: "ASP.NET Core breaking change"
+description: Report a change in ASP.NET Core that breaks something that worked in a previous version. Intended mostly for product-team use.
```
Comment (mkArtakMSFT) [inline .github/ISSUE_TEMPLATE/breaking-change.md:2]:
```suggestion
description: Report a change in ASP.NET Core that breaks something that worked in a previous version. This is Intended to be used by the product-team.
```

File: .github/ISSUE_TEMPLATE/breaking-change.md [hunk 2]
```diff
@@ -1,65 +1,80 @@
...
+title: "[Breaking change]: "
+labels: "Breaking change"
+assignees:
+ - gewarren
```
Comment (mkArtakMSFT) [inline .github/ISSUE_TEMPLATE/breaking-change.md:6]: Any specific reason for you to be auto-assigned to these announcements?

File: .github/ISSUE_TEMPLATE/breaking-change.md [hunk 3]
```diff
@@ -1,65 +1,80 @@
...
+    id: version
+    attributes:
+      label: Version
+      description: What version of .NET introduced the breaking change?
+      options:
+        - .NET 6 Preview 6
+        - .NET 6 Preview 7
+        - .NET 6 RC1
+        - .NET 6 RC2
+        - .NET 6 GA
```
Comment (mkArtakMSFT) [inline .github/ISSUE_TEMPLATE/breaking-change.md:25]: Should these only include RC1, RC2 and GA? I assume most of the preview7 breaking changes (if any) are already announced)

File: .github/ISSUE_TEMPLATE/breaking-change.md [hunk 4]
```diff
@@ -1,65 +1,80 @@
...
+      description: Describe the recommended action an affected user should take, such as workarounds or examples of code changes.
+    validations:
+      required: true
+  - type: dropdown
+    id: area
+    attributes:
+      label: Feature area
+      description: What feature area(s) does the change affect?
+      options:
+        - ASP.NET Core (Default)
```
Comment (mkArtakMSFT) [inline .github/ISSUE_TEMPLATE/breaking-change.md:72]: Is this field necessary at all, given that there is only this one option?

**Review by mkArtakMSFT — COMMENTED** [review]
Thanks for your PR, @gewarren.

One question I have is whether this is done across other repos too, or only here? And if only in this repo, then why?

**Comment by gewarren** [comment]
> Thanks for your PR, @gewarren.
> 
> One question I have is whether this is done across other repos too, or only here? And if only in this repo, then why?

We also have this form in the dotnet/docs repo [here](https://github.com/dotnet/docs/issues/new?assignees=gewarren&labels=breaking-change&template=breaking-change.yml&title=%5BBreaking+change%5D%3A+). Since we are linking to an [issuesof.net query](https://issuesof.net/?q=%20is:open%20-label:Documented%20is:issue%20(label:%22Breaking%20Change%22%20or%20label:breaking-change)%20(repo:dotnet%2Fdocs%20or%20repo:aspnet%2FAnnouncements)%20group:repo%20(label:%22:checkered_flag:%20Release%20.NET%206%22%20or%20label:6.0.0)%20sort:created-desc) for these two repos from the main .NET 6 breaking changes page, we need to have some consistency in the breaking change issues between these two repos. For example, we will label them with the type of breaking change (source incompatibility or binary incompatibility).

File: .github/ISSUE_TEMPLATE/breaking-change.md (hunk 4 above)
Comment (gewarren) [reply .github/ISSUE_TEMPLATE/breaking-change.md:72]: Probably not, I can remove it.

File: .github/ISSUE_TEMPLATE/breaking-change.md (hunk 2 above)
Comment (gewarren) [reply .github/ISSUE_TEMPLATE/breaking-change.md:6]: So I get notified to document it. I can leave it out if you prefer.

**Comment by gewarren** [comment]
cc @priyapurkayastha

**Comment by gewarren** [comment]
Changes were already applied.

**Pull request closed without merging**